datasets/kz/kz_companies/crawler.py

```diff
@@ -0,0 +1,120 @@
+from time import sleep
+
+from zavod import Context
+from zavod import helpers as h
+from rigour.urls import build_url
+
+
+BASE_URL = "https://data.egov.kz/datasets/view?index=gbd_ul"
+CRAWL_URL = "https://data.egov.kz/datasets/getdata"
+CACHE_DAYS = 30
+SLEEP_TIME = 1
+
+FORMDATA = {
+    "index": "gbd_ul",
+    "version": "v1",
+    "page": "1",
+    "count": "20",
+    "text": "",
+    "column": "id",
+    "order": "ascending",
+}
+
+FIELDS_MAPPING = {
+    "datereg": "incorporationDate",
+    "nameru": "name",
+    "addresskz": "address",
+    "namekz": "name",
+    "bin": "registrationNumber",
+    "addressru": "address",
+    "statuskz": "status",
+    "statusru": "status",
+    "okedru": "notes",
+    "okedkz": "notes",
+}
+
+
+HEADERS = {
+    "Referer": BASE_URL,
+    "X-Requested-With": "XMLHttpRequest",
+}
+
+
+def crawl_page(context: Context, page_number: int) -> int:
+    """
+    Crawls a single page of the Kazakhstan data portal and emits the data.
+    Args:
+        context: The context object for the current dataset.
+        page_number: The page number to crawl.
+    Returns:
+        The total number of pages.
+    """
+
+    formdata = FORMDATA.copy()
+    formdata["page"] = str(page_number)
+
+    data = context.fetch_json(
+        CRAWL_URL,
+        cache_days=CACHE_DAYS,
+        headers=HEADERS,
+        params=formdata,
+    )
```

Comment:
Could you add a data section to the metadata and then use `context.data_url` here instead of the constant in the file? That just makes sure we keep the metadata in sync with the crawler unless we really need something different.